src/ray/raylet/scheduling/fair_scheduling_queue_test.cc

```diff
@@ -0,0 +1,142 @@
+#include "ray/raylet/scheduling/fair_scheduling_queue.h"
+#include "ray/raylet/scheduling/test_util.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace ray {
+
+namespace raylet {
+
+using ::testing::_;
+
+class FairSchedulingQueueTest : public ::testing::Test {
+ public:
+  FairSchedulingQueue queue_;
+
+  Work TaskToWork(Task &task) { return std::make_tuple(task, nullptr, nullptr); }
+
+  void AssertNoLeaks() {
+    ASSERT_EQ(queue_.active_tasks_.size(), 0);
+    ASSERT_EQ(queue_.work_queue_.size(), 0);
+  }
+};
+
+TEST_F(FairSchedulingQueueTest, TestBasic) {
+  /*
+   * When tasks are queued, make sure we prioritize the shapes that aren't
+   * currently running.
+   */
+  Task task = CreateTask({{ray::kCPU_ResourceLabel, 4}});
+  Task task1 = CreateTask({{ray::kCPU_ResourceLabel, 2}});
+  Task task2 = CreateTask({{ray::kCPU_ResourceLabel, 1}});
+
+  queue_.MarkRunning(task);
+  queue_.MarkRunning(task);
+  queue_.MarkRunning(task1);
+
+  queue_.Push(TaskToWork(task));
+  queue_.Push(TaskToWork(task1));
+  queue_.Push(TaskToWork(task2));
+
+  std::vector<std::pair<const SchedulingClass, std::deque<Work>>> ordered(queue_.begin(),
+                                                                          queue_.end());
+
+  Task expected[] = {task2, task1, task};
+  ASSERT_EQ(ordered.size(), 3);
+  for (int i = 0; i < 3; i++) {
+    std::deque<Work> &deque = ordered[i].second;
+    ASSERT_EQ(deque.size(), 1);
+    const Work &work = deque.front();
+    const Task &t = std::get<0>(work);
+    ASSERT_EQ(t.GetTaskSpecification().TaskId(),
+              expected[i].GetTaskSpecification().TaskId());
+  }
+}
+
+TEST_F(FairSchedulingQueueTest, TestPushAll) {
+  /*
+   * When tasks are queued, make sure we prioritize the shapes that aren't
```

Comment:
This explanation doesn't seem right? Also, can you actually test this scenario? `When tasks are queued, make sure we prioritize the shapes that aren't running` with this test ? It looks like we only have a single shape task?